codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -489,8 +495,15 @@ class JsonSerializerGenerator(
                 rustBlock("match input") {
                     for (member in context.shape.members()) {
                         val variantName = symbolProvider.toMemberName(member)
-                        withBlock("#T::$variantName(inner) => {", "},", unionSymbol) {
-                            serializeMember(MemberContext.unionMember(context, "inner", member, jsonName))
+
+                        if (member.isTargetUnit()) {
+                            withBlock("#T::$variantName => {", "},", unionSymbol) {
+                                serializeMember(MemberContext.unionMember(context, "", member, jsonName))
```

Comment:
I'm surprised this works with the reference name being empty. Can we instead just short-circuit to serializing an empty object to avoid any future issues?

Reply:
Thanks for catching. Commit e4f6559 unconditionally adds a reference name `inner`, but in the case of the Unit type, nothing will be rendered due to our change to `RustWriter.jsonObjectWriter`. One issue is that we have also made a similar change around the member reference name to `RustWriter.serializeUnion` in `QuerySerializerGenerator` but we haven't identify a kind of change over there that we made in `RustWriter.jsonObjectWriter`. We'll see what test breaks in CI.

Reply:
Updated protocol tests for serializers and parser generators to wash out test failures. Fixes addressed in 45ef80d and 884155e.